src/libstore/tests/derivation.cc

```diff
@@ -11,8 +12,32 @@ class DerivationTest : public LibStoreTest
 {
 };
 
-#define TEST_JSON(NAME, STR, VAL, DRV_NAME, OUTPUT_NAME) \
-    TEST_F(DerivationTest, DerivationOutput_ ## NAME ## _to_json) {    \
+class CaDerivationTest : public DerivationTest
+{
+    void SetUp() override
+    {
+        experimentalFeatureSettings.set("experimental-features", "ca-derivations");
+    }
+    void TearDown() override
+    {
+        experimentalFeatureSettings.set("experimental-features", "");
+    }
+};
+
+class ImpureDerivationTest : public DerivationTest
+{
+    void SetUp() override
+    {
+        experimentalFeatureSettings.set("experimental-features", "impure-derivations");
+    }
+    void TearDown() override
+    {
+        experimentalFeatureSettings.set("experimental-features", "");
+    }
+};
```

Comment:
I'm a little concerned that concurrent mutation of these may make the tests flaky in the future.

Reply:
I don't like it either!